src/main/java/Database/playlistLibrary.java

```diff
@@ -1,4 +1,181 @@
 package Database;
 
-public class playlistLibrary {
+import Entities.Playlist;
+import Entities.Song;
+import Entities.User;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * Playlist database class in the Frameworks & Drivers layer that follows the singleton design pattern, and
+ * implements an interface in the Application Business Rules layer for dependency inversion purposes.
+ * Note: since this is a singleton class, the constructor is private, and it can NOT be created with the
+ * new keyword outside this class.
+ */
+public class playlistLibrary implements playlistAccessInterface {
+    private static final playlistLibrary PLAYLIST_LIBRARY =
+            new playlistLibrary("./src/main/java/Database/playlists.csv", userList.getInstance());
+    Map<Integer, Playlist> playlistDatabase;
+    String filepath;
+    songLibrary library;
+    userList users;
+    List<Integer> saved;
+
+    /**
+     * Global static method to retrieve the single instance of the playlistLibrary.
+     * @return the single instance of the playlist library database
+     */
+    public static playlistLibrary getInstance(){
+        return PLAYLIST_LIBRARY;
+    }
+
+    private playlistLibrary(String path, userList users){
+        this.filepath = path;
+        saved = new ArrayList<>();
+
+        // TODO: modify once songLibrary, userList implemented
+        library = new songLibrary();
+        this.users = users;
+
+        playlistDatabase = loadFile();
+    }
+
+    /**
+     * Method to read the .csv file from the instance variable path and return the saved playlists.
+     * @return a hashmap containing all playlists saved in the .csv file keyed with String names
+     */
+    private Map<Integer, Playlist> loadFile() {
+        Map<Integer, Playlist> p = new HashMap<>();
+
+        try {
+            Scanner in = new Scanner(new File(filepath));
+            while (in.hasNextLine()){
+                // assumes 5 columns, properly filled
+                String line = in.nextLine();
+                String[] data = line.split(",");
+                p.put(Integer.parseInt(data[0]), buildPlaylist(data));
```

Comment:
Yeah I've heard, I chose Scanner out of familiarity since it's been a while since I've used a BufferedReader and I was trying to get this done as quickly as possible. I'll keep this in mind when I come back to complete these classes once the remaining entites/databases are implemented.